torch/csrc/jit/serialization/pickler.cpp

```diff
@@ -603,7 +603,9 @@ WriteableTensorData getWriteableTensorData(
   result.tensor_ = tensor;
   result.size_ = tensor.storage().nbytes();
   // TODO HIP support
-  if (tensor.storage().device_type() == DeviceType::CUDA && to_cpu) {
+  if ((tensor.storage().device_type() == DeviceType::CUDA ||
+       tensor.storage().device_type() == DeviceType::XPU) &&
+      to_cpu) {
```

Comment:
This looks highly questionable. Maybe the correct fix is `tensor.storage().device_type() != DeviceType::CPU`

Reply:
We are not sure whether other devices, like HIP, XLA, already implemented all below tensor and storage APIs. Could you give more detailed suggestion here?

Reply:
They're indirectly supported via the call to `.cpu()`, which these support.

Reply:
Thanks for quick reply. We will follow your suggestion soon. Thanks.

Reply:
The code has been changed to `tensor.storage().device_type() != DeviceType::CPU` for more general cases